src/corehost/cli/deps_format.cpp

```diff
@@ -173,41 +173,51 @@ bool deps_json_t::perform_rid_fallback(rid_specific_assets_t* portable_assets, c
     
     for (auto& package : portable_assets->libs)
     {
-        pal::string_t matched_rid = package.second.rid_assets.count(host_rid) ? host_rid : _X("");
-        if (matched_rid.empty())
+        for (size_t asset_type_index = 0; asset_type_index < deps_entry_t::asset_types::count; asset_type_index++)
         {
-            if (rid_fallback_graph.count(host_rid) == 0)
+            auto& rid_assets = package.second[asset_type_index].rid_assets;
+            pal::string_t matched_rid = rid_assets.count(host_rid) ? host_rid : _X("");
+            if (matched_rid.empty())
             {
-                trace::warning(_X("The targeted framework does not support the runtime '%s'. Some native libraries from [%s] may fail to load on this platform."), host_rid.c_str(), package.first.c_str());
-            }
-            else
-            {
-                const auto& fallback_rids = rid_fallback_graph.find(host_rid)->second;
-                auto iter = std::find_if(fallback_rids.begin(), fallback_rids.end(), [&package](const pal::string_t& rid) {
-                    return package.second.rid_assets.count(rid);
-                });
-                if (iter != fallback_rids.end())
+                if (rid_fallback_graph.count(host_rid) == 0)
                 {
-                    matched_rid = *iter;
+                    trace::warning(_X("The targeted framework does not support the runtime '%s'. Some native libraries from [%s] may fail to load on this platform."), host_rid.c_str(), package.first.c_str());
+                }
+                else
+                {
+                    const auto& fallback_rids = rid_fallback_graph.find(host_rid)->second;
```

Comment:
Could you, instead of calling `count(host_rid)` on `rid_fallback_graph` to determine if the framework supports a runtime, call `find(host_rid)` and branch on it being found or not?

Reply:
I didn't cleanup existing code too much - but this is simple enough. Good catch!